core/string.cpp

```diff
@@ -45,4 +49,30 @@ std::map<DType, std::string> &dtypeNameRegistry() {
   return registry;
 }
 
+const std::string to_iso_date(const scipp::core::time_point &item,
+                              const std::string &unit) {
+  int64_t ts = item.time_since_epoch();
+
+  if (unit == "ns") {
+    int64_t conv = 1000000000;
+    // time representation of the timestamp
+    int64_t time = ts / conv;
+    auto tm = *std::gmtime(&time);
+    // nanoseconds part
+    std::chrono::duration<int64_t, std::nano> dur(ts);
+    auto ns =
+        std::chrono::duration_cast<std::chrono::nanoseconds>(dur).count() %
+        conv;
+    std::stringstream ss;
+    ss << std::put_time(&tm, "%FT%T.") << std::setw(9) << std::setfill('0')
+       << ns << std::endl;
+    return ss.str();
+  } else {
```

Comment:
Check for seconds? Scipp also supports, e.g., `us`. It is fine to not support everything in the formatting, but at least we should throw an exception instead of assuming it is seconds.